File: backend/app/models/book.js
const mongoose = require("mongoose");

const bookSchema = new mongoose.Schema({
  id: { type: String, required: true, unique: true },
  title: { type: String, required: true },
  author: { type: String, required: true },
  description: { type: String, required: false },
  dateAdded: { type: Date, default: Date.now },
  price: { type: Number, required: true },
  coverImageUrl: { type: String, required: false },
  category: { type: String, required: true },
});

const Book = mongoose.model("Book", bookSchema);

module.exports = Book;

File: frontend/lib/app/models/book.dart
import 'package:json_annotation/json_annotation.dart';
part 'book.g.dart';

@JsonSerializable()
class Book {
  String? id;
  String? title;
  String? author;
  String? description;
  DateTime? dateAdded;
  double? price;
  String? coverImageUrl;
  String? category;

  Book({
    this.id,
    this.title,
    this.author,
    this.description,
    this.dateAdded,
    this.price,
    this.coverImageUrl,
    this.category,
  });

     factory Book.fromJson(Map<String, dynamic> json) => _$BookFromJson(json);
  Map<String, dynamic> toJson() => _$BookToJson(this);
}